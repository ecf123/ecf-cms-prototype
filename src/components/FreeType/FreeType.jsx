import "./FreeType.scss";

const FreeType = ({
  freeTypeLabelText,
  freeTypePlaceHolderText,
  handleFreeTypeValue,
  name,
  inputClassName
}) => {
  return (
    <div className="free-type">
      <label className="free-type__label">{freeTypeLabelText}</label>
      <textarea
        // free-type__input is used for the design regular input
        // free-type__input--empty is used for the design of a red input that has empty input after submit is clickec
        className={inputClassName}
        type="text"
        // Will apply spellcheck to inputed text
        spellCheck="true"
        placeholder={freeTypePlaceHolderText}
        name={name}
        // Value of input will update as text is typed
        onChange={handleFreeTypeValue}
      ></textarea>
    </div>
  );
};

export default FreeType;
