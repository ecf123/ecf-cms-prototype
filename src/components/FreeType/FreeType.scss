@use "../../styles/_reset.scss" as *;
@use "../../styles/_variables.scss" as *;

.free-type {
  font-family: $font-secondary;
  font-style: normal;
  font-weight: $font-weight-regular;
  font-size: $font-size-small;

  &__label {
    line-height: 1.25rem;
    color: $color-blue-navy;
  }

  &__input {
    box-sizing: border-box;
    display: flex;
    flex-direction: row;
    align-items: flex-start;
    margin-top: 0.4rem;
    padding: 0.75rem 1rem;
    gap: 0.5rem;
    resize: none;
    width: 32rem;
    height: 9.25rem;
    line-height: 160%;
    color: $color-blue-navy;
    background: $color-white;
    border: 0.063rem solid $color-grey-light;
    box-shadow: 0rem 1px 2px $color-blue-dark;
    border-radius: 0.5rem;

    &--empty{
      box-sizing: border-box;
      display: flex;
      flex-direction: row;
      align-items: flex-start;
      margin-top: 0.4rem;
      padding: 0.75rem 1rem;
      gap: 0.5rem;
      resize: none;
      width: 32rem;
      height: 9.25rem;
      line-height: 160%;
      color: $color-blue-navy;
      background: $color-white;
      border: 0.063rem solid $color-red-normal;
      box-shadow: 0rem 1px 2px $color-red-normal;
      border-radius: 0.5rem;
      &:focus{
        outline: none;
      }
      &::placeholder {
        color: $color-grey-dark;
      }
    }
    &:focus{
      outline: none;
    }
    &::placeholder {
      color: $color-grey-dark;
    }
  }
}
