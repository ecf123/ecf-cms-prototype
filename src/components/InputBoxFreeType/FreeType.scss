@use "../../styles/_variables.scss" as *;
@use "../../styles/_typography.scss" as *;
@use "../../styles/_reset.scss" as *;

.freetype {

    &__label {
        width: 7.813rem;
        height: 1.25rem;

        font-family: 'Inter';
        font-style: normal;
        font-weight: 500;
        font-size: 0.875rem;
        line-height: 1.25rem;

        color: #344054;

        flex: none;
        order: 0;
        flex-grow: 0;
        }

  &__input {

    box-sizing: border-box;

    display: flex;
    flex-direction: row;
    align-items: flex-start;
    padding: 0.75rem 1rem;
    gap: 0.5rem;

    resize: none;

    width: 32rem;
    height: 9.25rem;

    background: #ffffff;

    border: 0.063rem solid #d0d5dd;

    box-shadow: 0rem 0.063rem 0.125rem rgba(16, 24, 40, 0.05);
    border-radius: 0.5rem;

    flex: none;
    order: 1;
    align-self: stretch;
    flex-grow: 0;

    ::placeholder {
        color: #667085;
    }
  }

textarea {
    width: 30rem;
    height: 7.188rem;

    font-family: 'Inter';
    font-style: normal;
    font-weight: 400;
    font-size: 1rem;
    line-height: 160%;

    color: #667085;

    flex: none;
    order: 0;
    flex-grow: 0;
    }
}
