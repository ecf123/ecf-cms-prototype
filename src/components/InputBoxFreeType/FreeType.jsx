import "./FreeType.scss";

const FreeType = ({
  labelText,
  placeHolderText,
  handleFreeTypeValue,
}) => {
  return (
    <div className="freetype">
      <label className="freetype__label">{labelText}</label>
      <textarea
        className="freetype__input"
        type="text"
        // Will apply spellcheck to inputed text
        spellCheck="true"
        placeholder={placeHolderText}
        // Value of input will update as text is typed
        onChange={handleFreeTypeValue}
      ></textarea>
    </div>
  );
};

export default FreeType;
