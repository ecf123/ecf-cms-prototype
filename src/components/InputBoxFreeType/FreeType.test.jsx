// FreeType.test.jsx
import { render, screen } from "@testing-library/react";
import FreeType from "./FreeType.jsx";
import { customRender } from "../../utils/testUtils";

it("Should match the snapshot, with the props", () => {
  const { container } = customRender(
    <FreeType
      freeTypelabelText={"label"}
      freeTypeplaceHolderText={"placeholder"}
    />
  );

  expect(container).toMatchSnapshot();
});

it("Should render the FreeType textarea", () => {
  // 1. Arrange
  render(<FreeType />);
  // 2. Act
  const textArea = screen.getByTestId("textarea");
  // 3. Assert
  expect(textArea).toBeInTheDocument();
  //comment
});

it("Should render the FreeType label", () => {
  render(<FreeType />);
  const label = screen.getByTestId("label");
  expect(label).toBeInTheDocument();
});
