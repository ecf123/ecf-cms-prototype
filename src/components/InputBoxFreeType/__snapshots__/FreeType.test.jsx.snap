// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match the snapshot, with the props 1`] = `
<div>
  <div
    class="freetype"
    data-testid="div"
  >
    <label
      class="freetype__label"
      data-testid="label"
    >
      label
    </label>
    <textarea
      class="freetype__input"
      data-testid="textarea"
      placeholder="placeholder"
      spellcheck="true"
      type="text"
    />
  </div>
</div>
`;

exports[`Should match the snapshot, without the props 1`] = `
<div>
  <div
    class="freetype"
    data-testid="div"
  >
    <label
      class="freetype__label"
      data-testid="label"
    />
    <textarea
      class="freetype__input"
      data-testid="textarea"
      spellcheck="true"
      type="text"
    />
  </div>
</div>
`;
